### DISCUSSION TITLE
What is the Successful Rate calculation logic？

### DISCUSSION THREAD

**Discussion by houbinglei** [apache/skywalking#10435] [Q&A]
env:
Observe Service Mesh through telemetry data collected from Envoy Access Log Service (ALS)
istio: 1.13.9
skywalking: 9.1.0

I find that when service_cpm is 0,  the success rate is also 0...，This results in a low success rate， is it a normal phenomenon？
the picture for the service_cpm:
![图片](https://user-images.githubusercontent.com/37292926/220829997-1a22888b-63ce-4966-aa76-e1e3ab366cbe.png)
the picture for the success_rate at the same time:
![图片](https://user-images.githubusercontent.com/37292926/220830106-fdd4ab73-dc88-43ab-9734-0cb2a340aa04.png)

**Comment by wu-sheng** [comment]
No request, then no rate. Isn't that?

**Reply by houbinglei** [reply]
Thanks! 
excuseme... i have another question... 
how can i  set service group on ALS services?
![图片](https://user-images.githubusercontent.com/37292926/221804416-37ba4cdf-20ab-4204-a6d8-b7eaea7fa129.png)

**Reply by wu-sheng** [reply]
The official showcase represents you how to control this. Read and try that.

**Reply by houbinglei** [reply]
oh~ 
i have seen this doc:
If the service name includes double colons (::), the literal string before the colons is taken as the group name. In the latest GraphQL query, the group name has been provided as an optional parameter.

 i use istio 1.16x with the conf:
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  meshConfig:
    defaultProviders:
      tracing:
      - "skywalking"
    enableTracing: true
    extensionProviders:
    - name: "skywalking"
      skywalking:
        service: tracing.istio-system.svc.cluster.local
        port: 11800
i need to set the service name with xxx::xxx ; but the service name is auto genarated ; and  there is no SkyWalking agent installed to the services, all the traces are generated by the sidecar proxies; could i customize the service name ? 
![企业微信截图_b6945850-cf8f-4cbf-8370-456ffc72b832](https://user-images.githubusercontent.com/37292926/222031250-78d661c7-4fde-4fc5-a00a-2c7e9e96179b.png)

**Reply by wu-sheng** [reply]
Again, read the showcase
- Repo, https://github.com/apache/skywalking-showcase
- Docs, https://skywalking.apache.org/docs/skywalking-showcase/next/readme/
